src/main/java/io/sdkman/broker/download/CandidateDownloadHandler.java

```diff
@@ -47,12 +58,44 @@ public void handle(Context ctx) {
                                             downloadResolver.resolve(downloads, p.name())
                                                     .ifPresentOrElse(v -> {
                                                         audit(details, p.id(), v.platform());
+                                                        writeChecksumHeaders(ctx, v.checksums());
                                                         ctx.redirect(302, v.url());
                                                     }, clientError(ctx, 404))),
                             clientError(ctx, 400));
         }, clientError(ctx, 404));
     }
 
+    private void writeChecksumHeaders(Context ctx,
+        Option<scala.collection.immutable.Map<String, String>> checksums) {
+        if(checksums == null || checksums.isEmpty()){
```

Comment:
The result is that you don't need to do this check because the option has already been dealt with.

Reply:
👍